File: newspark-art/python/steps/global_steps.py
from behave import given, when, then
from pages.login_page import LoginPage

@then('I should see an error message "{message}"')
def step_impl(context, message):
    error_message = context.page.locator('[data-testid="error-message"]')
    error_message.wait_for(state='visible', timeout=3000)
    assert error_message.text_content() == message, f"Expected error message '{message}'"

File: newspark-art/TypeScript/tests/step-definitions/global.steps.ts
import { Then } from '@cucumber/cucumber';
import { expect } from '@playwright/test';
import { CustomWorld } from '../support/world';

Then('I should see an error message {string}', async function(this: CustomWorld, message: string) {
  const errorMessage = this.page.locator('[data-testid="error-message"]');
  await errorMessage.waitFor({ state: 'visible', timeout: 3000 });
  const actualText = await errorMessage.textContent();
  expect(actualText).toBe(message);
});
